content/posts/google_sheets_as_apis.md

```diff
@@ -0,0 +1,148 @@
+4. **Create a Genezio project** starting from an existing template.
+5. **Update the project** with your personal data and credentials.
+6. **Test the project** to verify everything works.
+
+## Step 1: Enable the Google Sheets API
+
+First, enable the Google Sheets API for your project.
+
+1. **Go to the [Google Cloud Console](https://console.cloud.google.com/).**
+
+2. **Create a New Project** (if you don't have one):
+
+   - Click the project dropdown at the top of the page.
+   - Click **New Project**.
+   - Enter a project name (e.g., "Google Sheets Node.js Access").
+   - Click **Create**.
+
+3. **Select Your Project**:
+
+   - Ensure your new project is selected in the project dropdown.
+
+4. **Enable the Google Sheets API**:
+
+   - Navigate to **APIs & Services** > **Dashboard**.
+   - Click **Enable APIs and Services**.
+   - Search for **Google Sheets API**.
+   - Click on **Google Sheets API** and then click **Enable**.
+
+## Step 2: Create a Service Account and Generate a Key File
+
+A Service Account allows server-side applications to authenticate without user interaction.
+
+1. **Navigate to Service Accounts**:
+
+   - Go to **APIs & Services** > **Credentials**.
+   - Click on **Create Credentials** and select **Service Account**.
+
+2. **Create a Service Account**:
+
+   - **Service Account Name**: Enter a name (e.g., "sheets-access-account").
+   - **Service Account ID**: This will auto-populate based on the name.
+   - **Description**: Optional.
+   - Click **Create and Continue**.
+
+3. **Set Service Account Permissions**:
+
+   - For this tutorial, you don't need to grant any specific roles.
+   - Click **Continue**.
+
+4. **Grant Users Access to This Service Account**:
+
+   - Skip this step by clicking **Done**.
+
+5. **Generate a JSON Key File**:
+
+   - You should now see your service account listed.
+   - Click on the **Service Account** you just created.
+   - Navigate to the **Keys** tab.
+   - Click **Add Key** > **Create New Key**.
+   - Choose **JSON** as the key type.
+   - Click **Create**.
+   - A JSON key file will be downloaded to your computer (e.g., `service-account-key.json`).
+
+6. **Secure the Key File**:
+
+   - **Important**: Do not commit this file to version control. It's sensitive information.
+
+## Step 3: Share the Spreadsheet with the Service Account Email
+
+The Service Account acts like a virtual user. You need to share the spreadsheet with the Service Account's email address.
+
+1. **Get the Service Account Email**:
+
+   - In the **Details** tab of your Service Account, locate the **Email** field.
+   - It will look like: `your-service-account-name@your-project-id.iam.gserviceaccount.com`.
+
+2. **Share the Spreadsheet**:
+
+   - Open your Google Sheet in a browser.
+   - Click the **Share** button.
+   - Enter the Service Account email.
+   - Set the permission to **Editor** (or **Viewer** if you only need read access).
+   - Click **Send**.
+
+## Step 4: Create a new Genezio project
+
```

Comment:
Genezio simplifies server-side development by providing pre-built templates that can quickly be deployed to work with various services, including Google Sheets. This makes it a great option for developers looking to build scalable, serverless applications without dealing with complex configurations.

We have prepared a couple of templates for you, depending what technology you want to use. Pick one of them, then click on the "Deploy now" button from one of the following Github repositories:

- [Express.js - Google Spreadsheet App](https://github.com/Genez-io/google-sheets-as-apis-expressjs?tab=readme-ov-file#deploy)
- [Genezio Typescript - Google Spreadsheet App](https://github.com/Genez-io/google-sheets-as-apis-typesafe?tab=readme-ov-file#deploy)

**Pro tip**: With Genezio’s serverless infrastructure, you don’t need to worry about maintaining servers or scaling your backend, making it perfect for projects where you need to scale quickly.